Add nfschema plugin subwf

nvnieuwk opened the pull request:
This PR adds the `nf-schema` equivalent of `utils_nfvalidation_plugin` 

This PR cannot be merged before nf-schema v2.1.0 has been released. Which needs one more feature ATM (https://github.com/nextflow-io/nf-schema/issues/43)

nvnieuwk commented:
Tests are failing because it's using the unreleased `nf-schema` v2.1.0 (which I tested with a locally installed version of the plugin)

nvnieuwk commented on subworkflows/nf-core/utils_nfschema_plugin/tests/nextflow.config:
```diff
@@ -0,0 +1,8 @@
+}
+
+validation {
+    parametersSchema = "${projectDir}/subworkflows/nf-core/utils_nfschema_plugin/tests/nextflow_schema.json"
```
It's still possible to supply parameter schemas to these functions :) I'd advise against using multiple JSON schemas for one pipeline though since that complicates it a lot

maxulysse replied:
and for a meta pipeline?

nvnieuwk replied:
Wouldn't `workflow` be the object of the meta pipeline then? These subworkflows are also made to run at the start of a meta pipeline and not at the start of each part, right?

nvnieuwk replied:
Good point, I'll give in and add it as an input 😁

nvnieuwk replied:
Also seems like I mixed up both conversations here 😁

nvnieuwk commented:
@maxulysse and @adamrtalbot I added the workflow as a mandatory input to the subwf. I also added parameters schema as an optional input. This is optional because in normal pipeline I'd always advise to only set the schema using the `validation.parametersSchema` config value to keep it consistent for the whole pipeline. It would be wise though to use this input for meta pipelines.

nvnieuwk commented:
> What about the failing tests?

Seems like some other modules were failing, I updated the branch :p
